util/util.go

```diff
@@ -75,14 +81,31 @@ func ModifyImageTag(imageName, tagName string) (string, error) {
 	}
 
 	// update the image tag with tagName
-	namedTagged, err := reference.WithTag(namedRef, tagName)
+	namedTagged, err := reference.WithTag(namedRef, normalisedTagName)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to update image tag")
 	}
 
 	return reference.FamiliarString(reference.TagNameOnly(namedTagged)), nil
 }
 
+func KubernetesVersionIsValid(version string) bool {
+	return kubeReleaseRegex.MatchString(version)
+}
+
+func ImageTagIsValid(tagName string) bool {
+	return !ociTagAllowedChars.MatchString(tagName)
```

Comment:
FYI: i've been proposing in the future to block user supplied tags if they are not semver on kubeadm runtime.
this would not matter as much for immutable upgrades, but would matter if we attempt to handle vendor specific metadata in mutable upgrades.

Reply:
cc @rosti

Reply:
which tags are you referring to?

Reply:
semver's not strictly compatible with OCI, so would need to be careful with that. Can we cross that bridge when we come to it?

Reply:
i'm referring to the kubeadm tags for etcd and coredns, just commented over the OCI tag logic, which is not fully related.

Reply:
adding validation seems 👍, what kind of validation seems to be seen given that what @randomvariable mentioned stands unfortunately